### DISCUSSION TITLE
Sorting or Searching Role User using Spatie Role&Permission

### DISCUSSION THREAD

**Discussion by edikurniawan-dev** [rappasoft/laravel-livewire-tables#727] [Q&A]
How to sorting or searching by role name using package spatie role&permission?
Thank you

**Comment by carbtest** [comment]
`Column::make('Roles','roles.name')
            ->label(function($row){
                $roles = $row->roles()->get()->pluck('name')->toArray();
                return implode(", ", $roles);
            })->searchable(),`

But mine is not searchable either anyone have suggestions?

**Reply by lrljoe** [reply]
In the docs it discusses callbacks for the searchable and sortable methods.

You'll need to use the callbacks when you're using a label Column or want more control over how things are searchable/sortable
